File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math
def pn(a):
    if a == 1 or a == 0:
        return True
    for i in range(2, int(math.sqrt(a)+1)):
        if a % i == 0:
            return True
    return False
            

def solution(n, k):
    origin = n
    answer = 0
    l1 = []
    
    while True:
        if n < k:
            l1.append(n % k)
            break
        l1.append(n % k)
        n = n // k
    l1.reverse()
    l1.append(0)
    
    s = "0"
    
    for i in range(len(l1)):
        if l1[i] != 0:
            s += str(l1[i])
        else:
            print(int(s), i)
            if pn(int(s)) == False:
                print(True)
                answer += 1
            s = "0"
        
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;
import java.io.*;

class Solution {
    int answer = 0;
    
    private boolean isPrime(long x) {
        if (x <= 1) return false;
        if (x == 2) return true;
        if (x % 2 == 0) return false;

        for (long i = 3; i * i <= x; i += 2) {
            if (x % i == 0) return false;
        }
        return true;
    }

    private void calculate(String s) {
        long v = Long.parseLong(s);
        if (isPrime(v)) answer++;
    }
    
    public int solution(int n, int k) {
        String s1;
        StringBuilder sb = new StringBuilder();
        
        while (n > 0) {
            sb.append(n % k);
            n = n / k;
        }
        
        s1 = sb.reverse().toString();
                
        sb = new StringBuilder();
        
        
        for (int i = 0; i < s1.length(); i++) {
            if (s1.charAt(i) == '0' && sb.length() == 0) {
                continue;
            }
            
            if (s1.charAt(i) == '0') {
                calculate(sb.toString());
                sb = new StringBuilder();
                continue;
            }
            
            sb.append(s1.charAt(i));

            if (i == s1.length()-1) {
                calculate(sb.toString());
            }
        }
        
        
        return answer;
    }
}